### ISSUE TITLE
Insight Camel integration

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#39]
It is planned to integrate insight-camel from Fabric8 as a portable component that runs on Gravia. Storage is delegated to elastic search. Hawt.io plugin via Kibana

**Comment by jimmidyson** [comment]
You need to create an embedded Elasticsearch node to enable storage. This is done through a config admin factory that uses the `io.fabric8.elasticsearch-*` PIDs (you can create multiple nodes in the same container if you wish). So for example create an empty PID called `io.fabric8.elasticsearch-camel` & an embedded Elasticsearch will be created. The log & metrics storage engines will pick this up & data will be stored in there.

If you want to configure Elasticsearch (e.g. cluster name) then configure the PID appropriately with the normal Elasticsearch config, detailed at http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/setup-configuration.html#settings & http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/modules.html. You should be able to run with defaults without problems though.

**Comment by tdiesler** [comment]
Getting 

```
16:50:06,550 ERROR [io.fabric8.insight.elasticsearch-factory:2.0.0.SNAPSHOT] (MSC service thread 1-6) [io.fabric8.elasticsearch] The activate method has thrown an exception: java.lang.NoSuchFieldError: LUCENE_3_6
	at org.elasticsearch.Version.<clinit>(Version.java:45) [elasticsearch-1.4.2.jar:]
	at org.elasticsearch.node.internal.InternalNode.<init>(InternalNode.java:136) [elasticsearch-1.4.2.jar:]
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:159) [elasticsearch-1.4.2.jar:]
	at org.elasticsearch.node.NodeBuilder.node(NodeBuilder.java:166) [elasticsearch-1.4.2.jar:]
```

Depends on #194

**Comment by tdiesler** [comment]
Deferred to 2.2.0

**Comment by jimmidyson** [comment]
Just thinking about some changes to how camel insight works. Can I assume that every camel context is available through scr? Thinking of using service references rather than using the singleton camel container to trace/profile routes. This is definitely available in karaf, just checking if it is in wildfly before making the changes.

**Comment by tdiesler** [comment]
Every camel context is available through JMX. There is also a new integration point with Camel (i.e. additional to Container) that hooks into CamelContext creation that does not suffer from the issues I pointed out about Container - I can give you the details if needed. Gravia services or scr components are not available.

**Comment by jimmidyson** [comment]
Could you share the alternative to Camel Container? Problem with having each camel context available only through JMX is that they can be configured not to be exposed through JMX at all... (`disableJMX`) which means they're not guaranteed to be available there. Also, it is the `ManagedCamelContextMBean` that is available not the actual `CamelContext`. I need to be able to add interceptors to the route to aid with tracing (in addition to normal Camel `Tracer`) & this is not available via JMX.

**Comment by tdiesler** [comment]
Yes, good point. Have a look at CamelContextTrackerRegistry

**Comment by jimmidyson** [comment]
Sadly this isn't totally portable though right? In an OSGi world we have the problem of what happens if your camel context tracker is activated after camel contexts have already been created?

**Comment by tdiesler** [comment]
IMHO, it is a fundamental shortcoming that Camel does not have a notion of Runtime that governs CamelContext lifecycle. With CamelContextTrackerRegistry it is assumed that the container registers handlers as part of its bootstrap process i.e. before any user code is executed

**Comment by tdiesler** [comment]
We might want to go with [wildfly-monitor](https://github.com/rhq-project/wildfly-monitor)

**Comment by Jalaj28** [comment]
Hi ,

Can anyone help me with this error for lucene morelikethis function
Exception in thread "main" java.lang.NoSuchFieldError: LUCENE_42
	at testprog.Main.start(Main.java:47)
	at testprog.Main.main(Main.java:32)

**Comment by jamesnetherton** [comment]
@Jalaj28 the version constants are defined [here](https://lucene.apache.org/core/5_5_0/core/org/apache/lucene/util/Version.html), so you're likely using an out of date Lucene version.

You'll get a better response if you take this discussion to the Lucence mailing lists. We're more focussed on WildFly and Camel integration :smiley:
